File: binding/include/libqrgba64.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qrgba64.html
pub const qrgba64 = struct {

    /// New constructs a new QRgba64 object.
    pub fn New() QtC.QRgba64 {
        return qtc.QRgba64_new();
    }


    /// New2 constructs a new QRgba64 object.
    pub fn New2(param1: ?*anyopaque) QtC.QRgba64 {
        return qtc.QRgba64_new2(@ptrCast(param1));
    }


    pub fn FromRgba64(c: u64) QtC.QRgba64 {
        return qtc.QRgba64_FromRgba64(c);
    }

    pub fn FromRgba642(red: u16, green: u16, blue: u16, alpha: u16) QtC.QRgba64 {
        return qtc.QRgba64_FromRgba642(red, green, blue, alpha);
    }

    pub fn FromRgba(red: u8, green: u8, blue: u8, alpha: u8) QtC.QRgba64 {
        return qtc.QRgba64_FromRgba(red, green, blue, alpha);
    }

    pub fn FromArgb32(rgb: u32) QtC.QRgba64 {
        return qtc.QRgba64_FromArgb32(rgb);
    }

    pub fn IsOpaque(self: ?*anyopaque, ) bool {
        return qtc.QRgba64_IsOpaque(@ptrCast(self));
    }

    pub fn IsTransparent(self: ?*anyopaque, ) bool {
        return qtc.QRgba64_IsTransparent(@ptrCast(self));
    }

    pub fn Red(self: ?*anyopaque, ) u16 {
        return qtc.QRgba64_Red(@ptrCast(self));
    }

    pub fn Green(self: ?*anyopaque, ) u16 {
        return qtc.QRgba64_Green(@ptrCast(self));
    }

    pub fn Blue(self: ?*anyopaque, ) u16 {
        return qtc.QRgba64_Blue(@ptrCast(self));
    }

    pub fn Alpha(self: ?*anyopaque, ) u16 {
        return qtc.QRgba64_Alpha(@ptrCast(self));
    }

    pub fn SetRed(self: ?*anyopaque, _red: u16) void {
        qtc.QRgba64_SetRed(@ptrCast(self), _red);
    }

    pub fn SetGreen(self: ?*anyopaque, _green: u16) void {
        qtc.QRgba64_SetGreen(@ptrCast(self), _green);
    }

    pub fn SetBlue(self: ?*anyopaque, _blue: u16) void {
        qtc.QRgba64_SetBlue(@ptrCast(self), _blue);
    }

    pub fn SetAlpha(self: ?*anyopaque, _alpha: u16) void {
        qtc.QRgba64_SetAlpha(@ptrCast(self), _alpha);
    }

    pub fn Red8(self: ?*anyopaque, ) u8 {
        return qtc.QRgba64_Red8(@ptrCast(self));
    }

    pub fn Green8(self: ?*anyopaque, ) u8 {
        return qtc.QRgba64_Green8(@ptrCast(self));
    }

    pub fn Blue8(self: ?*anyopaque, ) u8 {
        return qtc.QRgba64_Blue8(@ptrCast(self));
    }

    pub fn Alpha8(self: ?*anyopaque, ) u8 {
        return qtc.QRgba64_Alpha8(@ptrCast(self));
    }

    pub fn ToArgb32(self: ?*anyopaque, ) u32 {
        return qtc.QRgba64_ToArgb32(@ptrCast(self));
    }

    pub fn ToRgb16(self: ?*anyopaque, ) u16 {
        return qtc.QRgba64_ToRgb16(@ptrCast(self));
    }

    pub fn Premultiplied(self: ?*anyopaque, ) QtC.QRgba64 {
        return qtc.QRgba64_Premultiplied(@ptrCast(self));
    }

    pub fn Unpremultiplied(self: ?*anyopaque, ) QtC.QRgba64 {
        return qtc.QRgba64_Unpremultiplied(@ptrCast(self));
    }

    pub fn OperatorunsignedLongLong(self: ?*anyopaque, ) u64 {
        return qtc.QRgba64_OperatorunsignedLongLong(@ptrCast(self));
    }

    pub fn OperatorAssign(self: ?*anyopaque, _rgba: u64) void {
        qtc.QRgba64_OperatorAssign(@ptrCast(self), _rgba);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QRgba64_Delete(@ptrCast(self));
    }
};

File: binding/src/libqrgba64.cpp
#include <QRgba64>
#include <qrgba64.h>
#include "libqrgba64.h"
#include "libqrgba64.hxx"

QRgba64* QRgba64_new() {
	 return new QRgba64();
}

QRgba64* QRgba64_new2(const QRgba64* param1) {
	 return new QRgba64(*param1);
}

QRgba64* QRgba64_FromRgba64(unsigned long long c) {
	return new QRgba64(QRgba64::fromRgba64(c));
}

QRgba64* QRgba64_FromRgba642(uint16_t red, uint16_t green, uint16_t blue, uint16_t alpha) {
	return new QRgba64(QRgba64::fromRgba64(red, green, blue, alpha));
}

QRgba64* QRgba64_FromRgba(unsigned char red, unsigned char green, unsigned char blue, unsigned char alpha) {
	return new QRgba64(QRgba64::fromRgba(red, green, blue, alpha));
}

QRgba64* QRgba64_FromArgb32(unsigned int rgb) {
	return new QRgba64(QRgba64::fromArgb32(rgb));
}

bool QRgba64_IsOpaque(const QRgba64* self) {
	return self->isOpaque();
}

bool QRgba64_IsTransparent(const QRgba64* self) {
	return self->isTransparent();
}

uint16_t QRgba64_Red(const QRgba64* self) {
	return self->red();
}

uint16_t QRgba64_Green(const QRgba64* self) {
	return self->green();
}

uint16_t QRgba64_Blue(const QRgba64* self) {
	return self->blue();
}

uint16_t QRgba64_Alpha(const QRgba64* self) {
	return self->alpha();
}

void QRgba64_SetRed(QRgba64* self, uint16_t _red) {
	self->setRed(_red);
}

void QRgba64_SetGreen(QRgba64* self, uint16_t _green) {
	self->setGreen(_green);
}

void QRgba64_SetBlue(QRgba64* self, uint16_t _blue) {
	self->setBlue(_blue);
}

void QRgba64_SetAlpha(QRgba64* self, uint16_t _alpha) {
	self->setAlpha(_alpha);
}

unsigned char QRgba64_Red8(const QRgba64* self) {
	return self->red8();
}

unsigned char QRgba64_Green8(const QRgba64* self) {
	return self->green8();
}

unsigned char QRgba64_Blue8(const QRgba64* self) {
	return self->blue8();
}

unsigned char QRgba64_Alpha8(const QRgba64* self) {
	return self->alpha8();
}

unsigned int QRgba64_ToArgb32(const QRgba64* self) {
	return self->toArgb32();
}

uint16_t QRgba64_ToRgb16(const QRgba64* self) {
	return self->toRgb16();
}

QRgba64* QRgba64_Premultiplied(const QRgba64* self) {
	return new QRgba64(self->premultiplied());
}

QRgba64* QRgba64_Unpremultiplied(const QRgba64* self) {
	return new QRgba64(self->unpremultiplied());
}

unsigned long long QRgba64_OperatorunsignedLongLong(const QRgba64* self) {
	return self->operator unsigned long long();
}

void QRgba64_OperatorAssign(QRgba64* self, unsigned long long _rgba) {
	self->operator=(_rgba);
}

void QRgba64_Delete(QRgba64* self) {
    delete self;
}

